libraries/libpensol/src/main/java/org/pentaho/reporting/libraries/pensol/PublishRestUtil.java

```diff
@@ -109,32 +114,30 @@ public int publishFile( String filePath, byte[] data, boolean overwriteIfExists
    *
    * @param repositoryPath    repository folder where to upload the file; must be separated with /
    * @param fileName          uploaded file's name; must not contain prohibited symbols
+   * @param fileProperties    the properties which should be applied to the file on repository
    * @param fileInputStream   file's data stream
-   * @param overwriteIfExists flag indicating an existing file should be overwritten
    * @return http response code
    * @see org.pentaho.platform.web.http.api.resources.RepositoryPublishResource
    */
-  public int publishFile( String repositoryPath, String fileName, InputStream fileInputStream,
-                          boolean overwriteIfExists ) throws IOException {
-
-    String url = baseUrl + REPO_FILES_IMPORT;
-
+  public int publishFile( String repositoryPath, String fileName, InputStream fileInputStream, Optional<Properties> fileProperties ) throws IOException {
```

Comment:
Same as above. Get rid of the optional, use the empty collection directly. Check for null via Objects.requireNonNull(..).